### PR TITLE
Quickly fail scaleup on all instance creation errors

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: cluster-autoscaler/clusterstate/clusterstate.go [hunk 1]
```diff
@@ -976,30 +976,30 @@ func (csr *ClusterStateRegistry) GetClusterSize() (currentSize, targetSize int)
 	return currentSize, targetSize
 }
 
-func (csr *ClusterStateRegistry) handleOutOfResourcesErrors(currentTime time.Time) {
+func (csr *ClusterStateRegistry) handleInstanceCreationErrorsErrors(currentTime time.Time) {
```
Comment (seh) [inline cluster-autoscaler/clusterstate/clusterstate.go]: Just to be sure, did you mean to repeat the word "Errors" in this function name?

Comment (losipiuk) [reply cluster-autoscaler/clusterstate/clusterstate.go]: Of course not. Thanks :)

**Comment by krzysztof-jastrzebski** [comment]
Fix tests.

**Comment by losipiuk** [comment]
Updated

**Pull request merged**